include/deal.II/lac/full_matrix.templates.h

```diff
@@ -1188,8 +1188,21 @@ FullMatrix<number>::determinant () const
                +(*this)(2,0)*(*this)(0,1)*(*this)(1,2)
                -(*this)(2,0)*(*this)(0,2)*(*this)(1,1));
     default:
-      Assert (false, ExcNotImplemented());
-      return 0;
+      // Compute the determinant using the Laplace expansion of the
+      // determinant. We expand along the last row.
+      // This is the same approach used to compute the determinant
+      // of Tensors.
+      number det = number();
+      for (unsigned int k=0; k<this->n_cols(); ++k)
+        {
+          FullMatrix<number> minorMatrix(this->n_cols() - 1, this->n_rows() - 1);
+          for (unsigned int i=0; i<this->n_rows() - 1; ++i)
+            for (unsigned int j=0; j<this->n_cols() - 1; ++j)
+              minorMatrix[i][j] = (*this)(i,j<k ? j : j+1);
+          const number cofactor = ((k % 2 == 0) ? number(-1.) : number(1.)) * minorMatrix.determinant();
+          det += (*this)(this->n_rows()-1, k) * cofactor;
+        }
+      return ((this->n_cols() % 2 == 0) ? number(1.) : number(-1.)) * det;
```

Comment:
Almost all configurations do have LAPACK, so I would also be fine with simply bailing out in case there is no LAPACK. (And yes indeed, I was thinking about the LU factorized version.)